### PR TITLE
ORV2-1506 Database Table Creation for OnRouteBC Special Authorization

### PR DESCRIPTION
**Pull request by gchauhan-aot**
# Description
 This PR includes changes to add onroutebc special authorization table to the database.
Fixes # (issue)
ORV2-1506
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A: Tested using revert and migration scripts
- [ ] Test B

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1518-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1518-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1518-dops.apps.silver.devops.gov.bc.ca/api)
- [Policy](https://onroutebc-1518-policy.apps.silver.devops.gov.bc.ca/api)
- [Scheduler](https://onroutebc-1518-scheduler.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: database/mssql/test/versions/v_35_1_test.sql [hunk 1]
```diff
@@ -1,7 +1,7 @@
 -- Test that the auth groups have been inserted correctly
 SET NOCOUNT ON
 
-SELECT COUNT(*) FROM $(DB_NAME).[access].[ORBC_USER_AUTH_GROUP_TYPE] 
+SELECT COUNT(*) FROM $(DB_NAME).[access].[ORBC_GROUP_ROLE]
```
Comment (praju-aot) [inline database/mssql/test/versions/v_35_1_test.sql:4]: Does the main test 35.sh file needs to be updated?

File: database/mssql/scripts/versions/v_37_ddl.sql [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+GO
+BEGIN TRANSACTION
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+CREATE TABLE [permit].[ORBC_SPECIAL_AUTH](
+	[ID] [int] IDENTITY(1,1) NOT NULL,
+	[COMPANY_ID] [int] NOT NULL,
+	[LCV] [char](1) NOT NULL,
```
Comment (praju-aot) [inline database/mssql/scripts/versions/v_37_ddl.sql:20]: Should there be a default value as the col is set as NOT NULL?

File: database/mssql/scripts/versions/v_37_ddl.sql
Comment (praju-aot) [inline database/mssql/scripts/versions/v_37_ddl.sql:1]: Please add description of columns

File: database/mssql/scripts/versions/v_37_ddl.sql [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	[DB_CREATE_TIMESTAMP] [datetime2](7) NOT NULL,
+	[DB_LAST_UPDATE_USERID] [varchar](63) NOT NULL,
+	[DB_LAST_UPDATE_TIMESTAMP] [datetime2](7) NOT NULL,
+ CONSTRAINT [PK_ORBC_NO_FEE_TYPE] PRIMARY KEY CLUSTERED 
+(
+	[NO_FEE_TYPE] ASC
+)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
+) ON [PRIMARY]
+GO
+
```
Comment (praju-aot) [inline database/mssql/scripts/versions/v_37_ddl.sql:90]: Missing initial insertion of data into ORBC_NO_FEE_TYPE

**Review by praju-aot — CHANGES_REQUESTED** [review]

File: database/mssql/test/versions/v_35_1_test.sql (hunk 1 above)
Comment (gchauhan-aot) [reply database/mssql/test/versions/v_35_1_test.sql:4]: nope, just this one.

File: database/mssql/scripts/versions/v_37_ddl.sql (hunk 2 above)
Comment (gchauhan-aot) [reply database/mssql/scripts/versions/v_37_ddl.sql:20]: Will set it

File: database/mssql/scripts/versions/v_37_ddl.sql
Comment (gchauhan-aot) [reply database/mssql/scripts/versions/v_37_ddl.sql:1]: Will do

File: database/mssql/scripts/versions/v_37_ddl.sql (hunk 3 above)
Comment (gchauhan-aot) [reply database/mssql/scripts/versions/v_37_ddl.sql:90]: Will add.

File: database/mssql/scripts/versions/v_37_ddl.sql (hunk 2 above)
Comment (gchauhan-aot) [reply database/mssql/scripts/versions/v_37_ddl.sql:20]: done

File: database/mssql/scripts/versions/v_37_ddl.sql
Comment (gchauhan-aot) [reply database/mssql/scripts/versions/v_37_ddl.sql:1]: done

File: database/mssql/scripts/versions/v_37_ddl.sql (hunk 3 above)
Comment (gchauhan-aot) [reply database/mssql/scripts/versions/v_37_ddl.sql:90]: done

File: database/mssql/scripts/versions/v_37_ddl.sql [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+-- Default values
+ALTER TABLE [permit].[ORBC_SPECIAL_AUTH] 
+   ADD CONSTRAINT [DF_ORBC_SPECIAL_AUTH_LCV] 
+   DEFAULT('N')
+   FOR [LCV]
+GO
+
+INSERT [permit].[ORBC_NO_FEE_TYPE] ([NO_FEE_TYPE], [DESCRIPTION], [CONCURRENCY_CONTROL_NUMBER], [DB_CREATE_USERID], [DB_CREATE_TIMESTAMP], [DB_LAST_UPDATE_USERID], [DB_LAST_UPDATE_TIMESTAMP]) VALUES (N'CA_GOVT', N'The government of Canada or any province or any territory', NULL, N'dbo', GETUTCDATE(), N'dbo', GETUTCDATE())
+INSERT [permit].[ORBC_NO_FEE_TYPE] ([NO_FEE_TYPE], [DESCRIPTION], [CONCURRENCY_CONTROL_NUMBER], [DB_CREATE_USERID], [DB_CREATE_TIMESTAMP], [DB_LAST_UPDATE_USERID], [DB_LAST_UPDATE_TIMESTAMP]) VALUES (N'MUN', N'A  municipality', NULL, N'dbo', GETUTCDATE(), N'dbo', GETUTCDATE())
```
Comment (praju-aot) [inline database/mssql/scripts/versions/v_37_ddl.sql]: Rename to 'MUNICPALITY'

**Review by praju-aot — CHANGES_REQUESTED** [review]

File: database/mssql/scripts/versions/v_37_ddl.sql (hunk 4 above)
Comment (gchauhan-aot) [reply database/mssql/scripts/versions/v_37_ddl.sql]: done

**Review by praju-aot — APPROVED** [review]

**Pull request merged**